compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1538,9 +1539,13 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                 precise_capturing
                     .iter()
                     .filter_map(|arg| match arg {
-                        ast::PreciseCapturingArg::Lifetime(lt) => Some(*lt),
-                        ast::PreciseCapturingArg::Arg(..) => None,
+                        PreciseCapturingArg::Lifetime(lt) => Some(*lt),
+                        PreciseCapturingArg::Arg(..) => None,
                     })
+                    // Add in all the lifetimes mentioned in the bounds. We will error
+                    // them out later, but capturing them here is important to make sure
+                    // they actually get resolved in resolve_bound_vars.
```

Comment:
how do we differentiate them later?

Reply:
We only record the ones in `use<...>` in `hir::OpaqueTy { precise_capturing_args }`.